src/integration/java/io/pinecone/integration/controlPlane/pod/ConfigureIndexTest.java

```diff
@@ -138,4 +144,20 @@ public void sizeIncrease() throws InterruptedException {
             assertEquals(podSpec.getPodType(), "p1.x2");
         });
     }
+
+    private static void waitUntilIndexStateIsReady(String indexName) throws InterruptedException {
+        int timeToWaitMs = 30000;
+        IndexModel index = controlPlaneClient.describeIndex(indexName);
+
+        while (index.getStatus().getState() != IndexModelStatus.StateEnum.READY && timeToWaitMs > 0) {
+            Thread.sleep(2000);
+            timeToWaitMs -= 2000;
+            System.out.println("waited 2000ms for index to upgrade, time left: " + timeToWaitMs);
+            System.out.println("System model state: " + index.getStatus());
+            index = controlPlaneClient.describeIndex(indexName);
+        }
+        if (!index.getStatus().getReady()) {
+            fail("Index " + indexName + " did not finish upgrading after " + timeToWaitMs + "ms");
```

Comment:
This is fail condition is fine, but I think the message is going to say 0ms in the message which isn't quite right.

Reply:
Ah doh, you're right nice catch.